### PR TITLE
[misc]: Refactor separate training pipeline runner

### PR DESCRIPTION
**Pull request by techverve**
### **What**
Decoupled the CLI execution logic from the legacy training pipelines by introducing a centralized runner script (`fastvideo/training/runner.py`).

### **Why**
To enforce separation of concerns and improve code maintainability. Previously, the legacy training pipelines coupled the class definitions directly with the script execution entry points. This architectural pattern resulted in redundant boilerplate across the codebase and made the pipeline modules harder to test and import in isolation.

### **How**
* Abstracted the `main()` execution blocks and CLI argument parsing out of the individual pipeline modules.
* Implemented a generic `runner.py` entry point that uses `importlib` to dynamically load the target pipeline at runtime via `--pipeline_class` and `--pipeline_module` flags.
* Refactored the underlying `finetune` and `distill` shell scripts to target the new unified runner.

MERGE WORKFLOW:
  1. Ensure pre-commit passes and you have at least 1 approval
  2. Comment /merge (or add the "ready" label) to enter the Merge Queue
  3. Full Test Suite runs automatically on a staging branch → auto-merge on success

ON-DEMAND TESTING (write access required):
  /test full       — Full Test Suite        /test ssim        — SSIM regression
  /test training   — Training pipeline      /test encoder     — Encoder tests
  /test transformer — Transformer tests     /test vae         — VAE tests
  /test kernel     — CUDA kernel tests      /test unit        — Unit tests
  See docs/contributing/pull_requests.md for all 17 test commands

## Purpose

Fixes https://linear.app/fastvideo/issue/FAS-215/training-separate-trainingpipeline-class-definition-and-runner-script

## Checklist

- [x] I ran `pre-commit run --all-files` and fixed all issues
- [x] I added or updated tests for my changes
- [ ] I updated documentation if needed
- [ ] I considered GPU memory impact of my changes

**For model/pipeline changes, also check:**
- [ ] I verified SSIM regression tests pass
- [ ] I updated the support matrix if adding a new model

### CODE REVIEW

**Comment by H1yori233** [comment]
Thanks for the contribution! This refactor makes sense. However, are many training scripts in the repo that invoke these pipeline files directly. Removing the existing entry points would break downstream compatibility. Could you keep backward-compatible entry points and update the in-repo scripts to use the new runner as well?

**Comment by techverve** [comment]
> Thanks for the contribution! This refactor makes sense. However, are many training scripts in the repo that invoke these pipeline files directly. Removing the existing entry points would break downstream compatibility. Could you keep backward-compatible entry points and update the in-repo scripts to use the new runner as well?

Hi @H1yori233 , I have addressed your suggestions could you please have a look again.
Thanks !

File: fastvideo/training/wan_training_pipeline.py [hunk 1]
```diff
@@ -63,6 +63,12 @@ def main(args) -> None:
 
 
 if __name__ == "__main__":
+    logger.warning(
+        "\n"
+        "================================================================================\n"
+        "[DEPRECATED]: Direct execution of this pipeline is deprecated!\n"
+        "Please use `fastvideo/training/runner.py` or the `fastvideo.training.runner.Runner` class instead.\n"
```
Comment (H1yori233) [inline fastvideo/training/wan_training_pipeline.py]: There is no `Runner` in `runner.py`, could you remove it to avoid confusion?

File: fastvideo/tests/training/test_runner.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        parser = FlexibleArgumentParser()
+        parser.add_argument("--pipeline-class",
+                            type=str,
+                            required=True)
+        parser.add_argument("--pipeline-module",
+                            type=str,
+                            required=True)
+        parser = TrainingArgs.add_cli_args(parser)
+        parser = FastVideoArgs.add_cli_args(parser)
+        return parser
```
Comment (H1yori233) [inline fastvideo/tests/training/test_runner.py]: This is duplicated in `runner.py`. I think could move it into `runner.py` so that the test and training environments stay consistent.

Comment (techverve) [reply fastvideo/tests/training/test_runner.py]: Makes sense, removed this duplication across other tests as well, can you please check now @H1yori233 ?

File: fastvideo/training/wan_training_pipeline.py (hunk 1 above)
Comment (techverve) [reply fastvideo/training/wan_training_pipeline.py]: removed

**Comment by H1yori233** [comment]
Thx, LGTM now.

**Comment by techverve** [comment]
> Thx, LGTM now.

Could you please approve it !
